src/Microsoft.Identity.Web.TokenCache/MsalAbstractTokenCacheProvider.cs

```diff
@@ -139,7 +134,7 @@ private async Task OnBeforeAccessAsync(TokenCacheNotificationArgs args)
             }
         }
 
-        private byte[] UnprotectBytes(byte[]? msalBytes)
+        private byte[]? UnprotectBytes(byte[]? msalBytes)
```

Comment:
The likely reason this regression was introduced is because it was unsure if this private method correctly handles null or not. Can we make it more obvious? Either rename to `UnprotectBytesIfNotNull` or by having the nulll check outside ?

Reply:
I just added a method comment, I think it's the clearest way. Otherwise rename would be UnprotectBytesIfNotNullAndEncryptedAndDataProtectionEnabled.